snap/snapenv/snapenv.go

```diff
@@ -106,16 +110,29 @@ func userEnv(info *snap.Info, home string) map[string]string {
 	return result
 }
 
-// envMap creates a map from the given environment string list, e.g. the
-// list returned from os.Environ()
-func envMap(env []string) map[string]string {
+var unsafeEnv = map[string]bool{
+	"HOSTALIASES": true,
+	"TMPDIR":      true,
+}
+
+const PreservedUnsafePrefix = "SNAP_SAVED_"
+
+// envMap creates a map from the given environment string list,
+// e.g. the list returned from os.Environ(). If preserveUnsafeVars
+// rename some variables that will be stripped out by the dynamic
+// linker executing the setuid snap-confine by prepending their names
+// with PreservedUnsafePrefix.
+func envMap(env []string, preserveUnsafeVars bool) map[string]string {
 	envMap := map[string]string{}
 	for _, kv := range env {
```

Comment:
```
if strings.HasPrefix(kv, PreservedUnsafePrefix) {
        continue
}
```

Reply:
It took me a moment to realize why you wanted this so I added a comment too.